core/cap-0072.md

```diff
@@ -224,61 +153,28 @@ pub trait StellarAccountInterface {
     fn update_thresholds(env: Env, low: Option<u32>, med: Option<u32>, high: Option<u32>);
 }
 ```
+### Classic transaction semantics
 
-### 'Classic' transaction semantics
-
-#### Ledger entry for additional account signers
-
-Every new or existing G-account on-chain may have up to 1 `ContractDataEntry` that contains additional signers (further referred to as 'additional signers entry'). The specification for the entry is as follows:
-
-- `contract` - the `SCAddress` corresponding to the account
-- `key` - `SCV_SYMBOL` `SCVal` with value `"SIGNERS" `
-- `durability` - `PERSISTENT`
-- `val` - `SCV_ACCOUNT_SIGNERS` `SCVal`
-
-`SCAccountSigners` payload may contain up to `MAX_SIGNERS` additional signers of type `SCAccountSigner` for the account (where `MAX_SIGNERS` is the same constant as one used for the `AccountEntry`, currently set to `20`). This new value type is introduced in the XDR changes above.
-
-`SCAccountSigner` may represent either a 'classic' account signer of type `Signer` (the same type as the one used for the `AccountEntry` itself), or a new kind of signer - `SCDelegatedAccountSigner`. For the `Signer` kind of signers only keys of type `SIGNER_KEY_TYPE_ED25519` will be allowed as of this CAP.
+A new type of G-account signer `SIGNER_KEY_TYPE_SC_DELEGATED` is added. It is supported in most of the contexts where the `Signer` struct is used, such as the `SetOptionsOp`, which thus allows adding/removing/updating the delegated signers. Only delegated signers with types convertible to the Soroban `Address` are supported, i.e. those with types `SC_ADDRESS_TYPE_ACCOUNT` and `SC_ADDRESS_TYPE_CONTRACT`. Otherwise, the operation will fail with `SET_OPTIONS_SIGNER_TYPE_NOT_SUPPORTED` error.
```

Comment:
I don't think this will cause issues, and maybe you're already aware, but wanted to mention that `RevokeSponsorshipOp` can take a `SignerKey` as input to transfer or revoke a signers sponsorship.

Reply:
I've missed this case, but since the operation looks for the existing signer key, I don't think there need to be any semantics updates or new errors here.